File: Lista_8/lista8_ex3.c
/* Exercício 3:
Faça uma função recursiva em C que calcula o elemento máximo em um vetor.
*/
#include <stdio.h>
#include <stdlib.h>

int maxRecursivo(int vet[], int n);

int main () {
int n=5, vet[] = {2, 3, 1, 5, 4};

printf("Elemento maximo: %d\n", maxRecursivo(vet, n));

return 0;
}

int maxRecursivo(int vet[], int n) {
    int aux;
    if(n == 1) {
        return vet[0];
    } else {
        aux = maxRecursivo(vet, n-1);
        if(aux > vet[n-1]) {
            return aux;
        }
        return vet[n-1];
    }
}

File: Lista_8/Lista8_ex3.py
def maximo_recursivo(vetor, n):
    if n == 1:
        return vetor[0]
    else:
        aux = maximo_recursivo(vetor, n-1)
        if aux > vetor[n-1]:
            return aux
        return vetor[n-1]

def main():
	vetor = [2, 3, 1, 5, 4]
	
	print("Elemento maximo: %d\n" % maximo_recursivo(vetor, len(vetor)))

if __name__ == "__main__":
	main()
